AwesomeCache/Cache.swift

```diff
@@ -2,268 +2,267 @@ import Foundation
-	/// - parameter cacheBlock:	This block gets called if there is no cached object or the cached object is already expired.
-	///                         The supplied success or failure blocks must be called upon completion.
-	///                         If the error block is called, the object is not cached and the completion block is invoked with this error.
+    /// The completion block is invoked as soon as the cacheBlock is finished and the object is cached.
+    ///
+    /// - parameter key:			The key to lookup the cached object
+    /// - parameter cacheBlock:	This block gets called if there is no cached object or the cached object is already expired.
+    ///                         The supplied success or failure blocks must be called upon completion.
+    ///                         If the error block is called, the object is not cached and the completion block is invoked with this error.
     /// - parameter completion: Called as soon as a cached object is available to use. The second parameter is true if the object was already cached.
-	public func setObjectForKey(key: String, cacheBlock: (CacheBlockClosure, ErrorClosure) -> Void, completion: (T?, Bool, NSError?) -> Void) {
-        
-		if let object = objectForKey(key) {
-			completion(object, true, nil)
-		} else {
-			let successBlock: CacheBlockClosure = { (obj, expires) in
-				self.setObject(obj, forKey: key, expires: expires)
-				completion(obj, false, nil)
-			}
-			
-			let failureBlock: ErrorClosure = { (error) in
-				completion(nil, false, error)
-			}
-			
-			cacheBlock(successBlock, failureBlock)
-		}
-	}
-	
-	
-	// MARK: Get object
-	
+    public func setObjectForKey(key: String, cacheBlock: (CacheBlockClosure, ErrorClosure) -> Void, completion: (T?, Bool, NSError?) -> Void) {
+        if let object = objectForKey(key) {
+            completion(object, true, nil)
+        } else {
+            let successBlock: CacheBlockClosure = { (obj, expires) in
+                self.setObject(obj, forKey: key, expires: expires)
+                completion(obj, false, nil)
+            }
+
+            let failureBlock: ErrorClosure = { (error) in
+                completion(nil, false, error)
+            }
+
+            cacheBlock(successBlock, failureBlock)
+        }
+    }
+
+
+    // MARK: Get object
+
     /// Looks up and returns an object with the specified name if it exists.
-    /// If an object is already expired, it is automatically deleted and `nil` will be returned.
+    /// If an object is already expired, `nil` will be returned.
     ///
     /// - parameter key: The name of the object that should be returned
+    /// - parameter returnExpiredObjectIfPresent: If set to `true`, an expired 
+    ///             object may be returned if present. Defaults to `false`.
     ///
     /// - returns: The cached object for the given name, or nil
-	public func objectForKey(key: String) -> T? {
-        return self.objectForKey(key, lockAlreadyHeld: false)
```

Comment:
Thanks for taking a look :) 

The tests (including the one you added) still pass and I have a version of this PR running in one of my apps in production. 
I've moved to use `add(object: CacheObject, key: String)` and `read(key: String) -> CacheObject?` (both private and not using a queue, thus not thread safe). The two methods are only invoked from within dispatch blocks.